nan.h

```diff
@@ -1654,12 +1661,12 @@ class Callback {
       Execute(progress);
   }
 
-  void SendProgress_(const char *data, size_t size) {
-    char *new_data = new char[size];
+  void SendProgress_(T *data, size_t size) {
+    T *new_data = new T[size];
     memcpy(new_data, data, size);
```

Comment:
It will unfortunstely not break compilation. The problem is that it is undefined behavior unless src is trivially copy constructible. That template could be used in a static assert to ensure a compilation error if invoked with a bad argument, but since said template is not available it cannot be used directly.

On June 30, 2016 6:18:52 PM GMT+03:00, "Robert Jefe Lindstädt" <notifications@github.com> wrote:
>> @@ -1654,12 +1661,12 @@ class Callback {
>>        Execute(progress);
>>    }
>>  
>> -  void SendProgress_(const char *data, size_t size) {
>> -    char *new_data = new char[size];
>> +  void SendProgress_(T *data, size_t size) {
>> +    T *new_data = new T[size];
>>      memcpy(new_data, data, size);
>
>I will likely have questions about this at some later point, if google
>doesn't yield enough information. I assume trying to compile with C++03
>will break it, hence makes it testable?
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/588/files/85aa1f0be5361cd79a2c85f75cb6296188223c87#r69152597